### PR TITLE
feat: Add StatusIcon, redesign styling for statuses

### PR DESCRIPTION
**Pull request by wackerow**
## Description
Reworks the new `ProofStatus` component
- Adds `StatusIcon` component with `status` variants 
- Usage: `<StatusIcon status="proved | proving | queued" />` with optional `className` styling

Creates new `DownloadButton` component
- Uses `Button` styling as base
- When "proved", button active
- If "proved" but no `proof` binary found, button will be disabled (shouldn't happen?)
- When "proving", tooltip describes, no onClick, and icon is animated with "pulse"
- When "queued", tooltip describes, no onClick

`downloadProof` function added, and attached to active Download button mentioned above

Includes proposal for design change:
- Use gray dashed stroke box for "queued"
- Use gray solid stroke box for "proving"
- Use green/primary solid stroke box for "proved"

This feels more additive as the process progresses. Gray and dashed > gains solid stroke > gains green color. Versus existing which is gray and solid > now green but regresses with dashed stroke > still green, gains solid stroke

image
image
image

### CODE REVIEW

File: components/DownloadButton.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        <span className={labelClassName}>Download proof</span>
+      </Button>
+    )
+
+  if (proof_status === "proving")
+    return (
+      <Tooltip
+        content={`${teamName} currently generating proof for this block`}
+      >
+        <Button size="icon" variant="outline" asChild>
```
Comment (wackerow) [inline components/DownloadButton.tsx:50]: @pettinarip See here for the fix to your comment from a previous PR... This prevents the repetition of using the ui/button styles as raw string here by just using asChild

File: lib/proofs.ts [hunk 2]
```diff
@@ -156,3 +156,24 @@ export const sortProofsStatusAndTimes = (a: Proof, b: Proof) => {
...
+  const { proof: binary, proof_id, block_number, team, machine_id } = proof
+
+  if (!binary) {
+    console.warn("Download failed - no proof binary found")
+    console.warn(`Block: ${block_number}, Proof ID: ${proof_id}`)
+    return
+  }
+
+  const teamName = team?.team_name ? team.team_name : machine_id.split("-")[0]
+  const suggestedFilename = `${block_number}_${teamName}_${proof_id}.bin`
```
Comment (pettinarip) [inline lib/proofs.ts:170]: Fine for me at this early point but is the `.bin` used in this context? or should we just download the file without an extension?

File: components/StatusIcon.tsx
Comment (pettinarip) [inline components/StatusIcon.tsx:1]: ✨

File: components/DownloadButton.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
+"use client"
```
Comment (pettinarip) [inline components/DownloadButton.tsx:1]: why do we need the client directive?

File: components/DownloadButton.tsx (hunk 1 above)
Comment (pettinarip) [reply components/DownloadButton.tsx:50]: Great! that works well.

**Review by pettinarip — APPROVED** [review]
LGTM! nice refactors

**Pull request merged**